spec/workspace-spec.js

```diff
@@ -1871,6 +1871,26 @@ i = /test/; #FIXME\
     })
   })
 
+  describe('::saveFocusedPaneItem', () => {
+    let editor = null
+    beforeEach(() =>
+      waitsForPromise(() => atom.workspace.open('sample.js').then(o => { editor = o }))
+    )
+
+    it("calls the focused item's save method", () => {
+      spyOn(editor, 'save')
+      spyOn(atom.workspace, 'getFocusedPane').andReturn(atom.workspace.getActivePane())
+      atom.workspace.saveFocusedPaneItem()
+      expect(editor.save).toHaveBeenCalled()
+    })
+    it("doesn't save the active editor if it's not focused", () => {
+      spyOn(editor, 'save')
+      spyOn(atom.workspace, 'getFocusedPane').andReturn(null)
+      atom.workspace.saveFocusedPaneItem()
+      expect(editor.save).not.toHaveBeenCalled()
+    })
+  })
+
```

Comment:
I think it'd be good to actually use the real implementation of `getFocusedPane` in these tests, since that's where some of the main complexity lies. You'd attach the workspace element to the DOM, but we often do that in order to test non-trivial focus stuff.